tests/python/relax/test_transform_remove_unused_parameters.py

```diff
@@ -97,5 +117,88 @@ def func(A: R.Tensor(["m", "n"], "float32")) -> R.Tensor(["m", "n"], "float32"):
     Expected = Before
 
 
+class TestRemoveExtraPrimVariables(BaseCompare):
+    """Remove parameters that only serve to define existing symbolic variables
+
+    If a `R.Prim` parameter provies a definition of a symbolic
+    variable, but that symbolic variable can be determined from a
+    different parameter, then the `R.Prim` parameter can be removed.
+    """
+
+    @I.ir_module
+    class Before:
+        @R.function
+        def main(A: R.Tensor(["m", "n"], "float32")) -> R.Tensor(["m", "n"], "float32"):
+            m = T.int64()
+            n = T.int64()
+            return Before.func(A, R.prim_value(m), R.prim_value(n))
+
+        @R.function(private=True)
+        def func(
+            A: R.Tensor(["m", "n"], "float32"), _m: R.Prim(value="m"), _n: R.Prim(value="n")
+        ) -> R.Tensor(["m", "n"], "float32"):
+            m = T.int64()
+            n = T.int64()
+            zeros = R.zeros(R.shape([m, n]), dtype="float32")
+            out = R.add(A, zeros)
+            return out
+
+    @I.ir_module
+    class Expected:
+        @R.function
+        def main(A: R.Tensor(["m", "n"], "float32")) -> R.Tensor(["m", "n"], "float32"):
+            return Expected.func(A)
+
+        @R.function(private=True)
+        def func(A: R.Tensor(["m", "n"], "float32")) -> R.Tensor(["m", "n"], "float32"):
+            m = T.int64()
+            n = T.int64()
+            zeros = R.zeros(R.shape([m, n]), dtype="float32")
+            out = R.add(A, zeros)
+            return out
+
+
+class TestRemoveExtraShapeVariables(BaseCompare):
+    """Remove parameters that only serve to define existing symbolic variables
+
+    If a `R.Shape` parameter provies a definition of a symbolic
```

Comment:
```suggestion
    If a `R.Shape` parameter provides a definition of a symbolic
```
typo :stuck_out_tongue_winking_eye:

Reply:
Thank you, and fixed!